### PR TITLE
25 debugging techniques

### PR DESCRIPTION
**Pull request by niksseif**
# Curriculum PR Template

### Does this PR close any issues?
*This PR closes [#25](https://github.com/turingschool/curriculum-site/issues/25)*

### Description

- [x] Revised the learning goals to emphasize general programming concepts.
- [x] Added detailed explanations of stack traces and common errors for JavaScript.
- [x] Enhanced the section on reading stack traces with specific examples and breakdowns for JS.
- [x] Improved grammar, clarity, and consistency across the entire lesson.
- [x] Ensured all code snippets and error messages are properly formatted for better readability.

### What questions do you have/what do you want feedback on? (optional)
- I am not sure if it is necessary to create a JavaScript exercise with tests specifically for this lesson? So far, I haven't added many examples for JavaScript—just a few error examples and their meanings, but no JavaScript code. Looking forward to any suggestions.

### CODE REVIEW

File: module1/lessons/index.md [hunk 1]
```diff
@@ -36,7 +36,7 @@ title: Module 1 - Lessons
 
 ## Debugging
 * [Async Intro to Debugging](./debugging_prework)
-<!-- * [Debugging Techniques](./debugging_techniques) -->
+* [Debugging Techniques](./debugging_techniques)
```
Comment (hfaerber) [inline module1/lessons/index.md:39]: I think this line is linking this lesson to itself, right?  I would remove this line.

**Review by hfaerber — CHANGES_REQUESTED** [review]
I like the detailed JS explanations you included Nik.  
Could you also include something generic about how JS developers often use console.log() to get visibility into their code and the data they're working with?  Maybe link to [mdn](https://developer.mozilla.org/en-US/docs/Web/API/console/log_static)?

File: module1/lessons/index.md (hunk 1 above)
Comment (niksseif) [reply module1/lessons/index.md:39]: @hfaerber this is the index file, which is a link to the lesson I made. If I remove it, then it won't show up on our curriculum site.

**Comment by niksseif** [comment]
@hfaerber - I've added context about console.log here e5c995e1fe3735a0afd87e7f06cf5be29bf9f8da

**Pull request merged**